src/utils.rs

```diff
@@ -1,24 +1,124 @@
 use protobuf::RepeatedField;
+use serde_derive::Serialize;
 
 use crate::onnx;
 use crate::onnx::OperatorSetIdProto;
 use crate::onnx::ValueInfoProto;
 use std::collections::HashMap;
 use std::convert::From;
 use std::convert::Into;
+use std::fmt::Display;
 use std::str::from_utf8;
 use thiserror::Error;
 
 /* Minimum size of a buffer you can create with wgpu. Creating buffers smaller than this leads to panic "Validation
 * error: buffer binding size X is less than minimum 64" in Device::create_bind_group */
-const MINIMUM_BUFFER_SIZE: i64 = 64;
+const MINIMUM_BUFFER_SIZE: u64 = 64;
 
-pub fn len(dims: &[i64]) -> i64 {
-    dims.iter().product::<i64>()
+#[derive(Debug, Serialize, Clone)]
+#[serde(transparent)]
+pub struct Shape(pub Vec<u64>);
```

Comment:
```suggestion
pub struct Shape {
 pub dim: Vec<u64>
};
```

I think Shape is the right terminology. 
I propose that we use the same struct definition layout as ndarray: https://docs.rs/ndarray/latest/src/ndarray/shape_builder.rs.html#9-14 in a way that if we want to add field to the struct in the future we can :)